bestbook/views/__init__.py

```diff
@@ -222,6 +222,14 @@ def persist_observation(cls, data):
             else:
                 all_observations[key] = value
 
+        existing_observations = Observation.get_all(username=data["username"],
+                                                    book_id=book.id)
+        
+        for elem in existing_observations:
+            aspect = Aspect.get(id=elem.aspect_id)
+            if aspect.label not in all_observations.keys():
```

Comment:
```suggestion
            if aspect.label not in all_observations:
```
`dict.keys()` is almost never needed in modern Python.

Reply:
Updated.  Thanks for all of your help with Python, @cclauss!